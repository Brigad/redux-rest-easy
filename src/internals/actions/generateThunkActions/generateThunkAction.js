const generateThunkAction = (actionType, resourceName) => normalizedURL => ({
  type: actionType,
  url: normalizedURL,
  resourceName,
});

export default generateThunkAction;
