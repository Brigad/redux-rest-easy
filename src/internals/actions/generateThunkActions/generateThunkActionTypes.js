import generateActionType from '../../utils/generateActionType';

const generateThunkActionTypes = resourceName => ({
  INVALIDATE_REQUEST: generateActionType(
    resourceName,
    '@invalidate',
    'INVALIDATE_REQUEST',
  ),
});

export default generateThunkActionTypes;
