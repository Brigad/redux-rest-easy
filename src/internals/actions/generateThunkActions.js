import generateThunk from '../action-creator/generateThunk';
import generateThunkAction from './generateThunkActions/generateThunkAction';
import generateThunkActionTypes from './generateThunkActions/generateThunkActionTypes';

const generateThunkActions = (
  resourceName,
  cacheLifetime,
  actionsConfig,
  actionName,
) => {
  const { INVALIDATE_REQUEST } = generateThunkActionTypes(resourceName);

  return {
    [actionName]: {
      perform: generateThunk(
        resourceName,
        cacheLifetime,
        actionsConfig,
        actionName,
      ),
      invalidate: generateThunkAction(INVALIDATE_REQUEST, resourceName),
    },
  };
};

export default generateThunkActions;
