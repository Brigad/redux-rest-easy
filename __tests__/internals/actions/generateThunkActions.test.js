import generateThunkActions from '../../../src/internals/actions/generateThunkActions';

describe('generateThunkActions', () => {
  test('only path', () => {
    const resourceName = 'fruits';
    const actionName = 'eat';
    const cacheLifetime = 0;
    const actionsConfig = {
      [actionName]: {
        method: 'GET',
        url: 'https://api.co/fruits',
      },
    };

    expect(
      generateThunkActions(
        resourceName,
        cacheLifetime,
        actionsConfig,
        actionName,
      ),
    ).toMatchSnapshot();
  });
});
