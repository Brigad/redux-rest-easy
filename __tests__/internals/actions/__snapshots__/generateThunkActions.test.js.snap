// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateThunkActions only path 1`] = `
Object {
  "eat": Object {
    "invalidate": [Function],
    "perform": [Function],
  },
}
`;
